{
 "metadata": {
  "name": "",
  "signature": "sha256:c3938fc01903938be79a12b32c478a0907c75a3fa12088d9766563aae4aed59b"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "Lamprey Transcriptome Analysis"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "```\n",
      "Camille Scott [camille dot scott dot w @gmail.com] [@camille_codon]\n",
      "\n",
      "camillescott.github.io\n",
      "\n",
      "Lab for Genomics, Evolution, and Development\n",
      "Michigan State University\n",
      "```"
     ]
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "About"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "This notebook is the entry point for the [Petromyzon marinus](http://nas.er.usgs.gov/queries/FactSheet.aspx?speciesID=836) (sea lamprey) de novo transcriptome analysis. This entry notebook contains links for the others, and code to collect and format data for the other notebooks. It should be run before all other notebooks in order to generate the requisite data."
     ]
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Contents"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "1. [Transcript Analysis Notebook](petmar-transcripts.ipynb)\n",
      "2. [Tissue Analysis Notebook](petmar-tissues.ipynb)\n",
      "3. [Protein Analysis Notebook](petmar-proteins.ipynb)\n",
      "4. [Taxonomic Analysis Notebook](petmar-taxonomy.ipynb)"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "from petmar_common import *"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 69
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Metadata"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "metadata = configtools.get_cfg('../../metadata.ini', '../../metadata.spec.ini')"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 70
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "outfmt6 = ['sseqid', 'pident', 'length', 'mismatch', 'gapopen', 'qstart', \\\n",
      "           'qend', 'sstart', 'send', 'evalue', 'bitscore']"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 71
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def wdir(fn):\n",
      "    return os.path.join('../../work/', fn)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 72
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def len_func(fn):\n",
      "    fn = wdir(fn).rstrip('.gz')\n",
      "    N = !grep -c \">\" {fn}\n",
      "    return int(N[0])"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 73
    },
    {
     "cell_type": "heading",
     "level": 3,
     "metadata": {},
     "source": [
      "Databases"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "db_metadata = pd.concat([pd.DataFrame(metadata['urls'].dict()).transpose(), pd.DataFrame(metadata['queries'].dict()).transpose()])\n",
      "db_metadata['size'] = db_metadata.dest.apply(len_func)\n",
      "db_metadata['fn'] = db_metadata.dest.apply(lambda s: s.rstrip('.gz'))\n",
      "db_metadata"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>db_type</th>\n",
        "      <th>dest</th>\n",
        "      <th>name</th>\n",
        "      <th>q_type</th>\n",
        "      <th>terms</th>\n",
        "      <th>url</th>\n",
        "      <th>size</th>\n",
        "      <th>fn</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>assembly</th>\n",
        "      <td> nucl</td>\n",
        "      <td>      lamp03.fasta.gz</td>\n",
        "      <td> lamp03</td>\n",
        "      <td>     NaN</td>\n",
        "      <td>                            NaN</td>\n",
        "      <td> http://athyra.ged.msu.edu/~cswelcher/lamprey/l...</td>\n",
        "      <td> 756840</td>\n",
        "      <td>      lamp03.fasta</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>danRer_pep</th>\n",
        "      <td> prot</td>\n",
        "      <td>     danRer.pep.fa.gz</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "      <td>                            NaN</td>\n",
        "      <td> ftp://ftp.ensembl.org/pub/release-75/fasta/dan...</td>\n",
        "      <td>  43153</td>\n",
        "      <td>     danRer.pep.fa</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>homSap_pep</th>\n",
        "      <td> prot</td>\n",
        "      <td>     homSap.pep.fa.gz</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "      <td>                            NaN</td>\n",
        "      <td> ftp://ftp.ensembl.org/pub/release-74/fasta/hom...</td>\n",
        "      <td> 104763</td>\n",
        "      <td>     homSap.pep.fa</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>musMus_pep</th>\n",
        "      <td> prot</td>\n",
        "      <td>     musMus.pep.fa.gz</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "      <td>                            NaN</td>\n",
        "      <td> ftp://ftp.ensembl.org/pub/release-75/fasta/mus...</td>\n",
        "      <td>  52165</td>\n",
        "      <td>     musMus.pep.fa</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>petMar2</th>\n",
        "      <td> nucl</td>\n",
        "      <td>        petMar2.fa.gz</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "      <td>                            NaN</td>\n",
        "      <td> ftp://ftp.ensembl.org/pub/release-75/fasta/pet...</td>\n",
        "      <td>  25006</td>\n",
        "      <td>        petMar2.fa</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>petMar2_cdna</th>\n",
        "      <td> nucl</td>\n",
        "      <td>   petMar2.cdna.fa.gz</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "      <td>                            NaN</td>\n",
        "      <td> ftp://ftp.ensembl.org/pub/release-75/fasta/pet...</td>\n",
        "      <td>  11489</td>\n",
        "      <td>   petMar2.cdna.fa</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>petMar2_cds</th>\n",
        "      <td> nucl</td>\n",
        "      <td>    petMar2.cds.fa.gz</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "      <td>                            NaN</td>\n",
        "      <td> ftp://ftp.ensembl.org/pub/release-75/fasta/pet...</td>\n",
        "      <td>  11442</td>\n",
        "      <td>    petMar2.cds.fa</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>petMar2_ncrna</th>\n",
        "      <td> nucl</td>\n",
        "      <td>  petMar2.ncrna.fa.gz</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "      <td>                            NaN</td>\n",
        "      <td> ftp://ftp.ensembl.org/pub/release-75/fasta/pet...</td>\n",
        "      <td>   2652</td>\n",
        "      <td>  petMar2.ncrna.fa</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>petMar2_pep</th>\n",
        "      <td> prot</td>\n",
        "      <td>    petMar2.pep.fa.gz</td>\n",
        "      <td>    NaN</td>\n",
        "      <td>     NaN</td>\n",
        "      <td>                            NaN</td>\n",
        "      <td> ftp://ftp.ensembl.org/pub/release-75/fasta/pet...</td>\n",
        "      <td>  11442</td>\n",
        "      <td>    petMar2.pep.fa</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Myxinidae_pep</th>\n",
        "      <td> prot</td>\n",
        "      <td>    Myx.pep.all.fa.gz</td>\n",
        "      <td>    NaN</td>\n",
        "      <td> uniprot</td>\n",
        "      <td>                  taxonomy:7762</td>\n",
        "      <td>                                               NaN</td>\n",
        "      <td>    990</td>\n",
        "      <td>    Myx.pep.all.fa</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>braFlo_pep</th>\n",
        "      <td> prot</td>\n",
        "      <td> braFlo.pep.all.fa.gz</td>\n",
        "      <td>    NaN</td>\n",
        "      <td> uniprot</td>\n",
        "      <td> organism:7739+AND+keyword:1185</td>\n",
        "      <td>                                               NaN</td>\n",
        "      <td>  28544</td>\n",
        "      <td> braFlo.pep.all.fa</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 28,
       "text": [
        "              db_type                  dest    name   q_type  \\\n",
        "assembly         nucl       lamp03.fasta.gz  lamp03      NaN   \n",
        "danRer_pep       prot      danRer.pep.fa.gz     NaN      NaN   \n",
        "homSap_pep       prot      homSap.pep.fa.gz     NaN      NaN   \n",
        "musMus_pep       prot      musMus.pep.fa.gz     NaN      NaN   \n",
        "petMar2          nucl         petMar2.fa.gz     NaN      NaN   \n",
        "petMar2_cdna     nucl    petMar2.cdna.fa.gz     NaN      NaN   \n",
        "petMar2_cds      nucl     petMar2.cds.fa.gz     NaN      NaN   \n",
        "petMar2_ncrna    nucl   petMar2.ncrna.fa.gz     NaN      NaN   \n",
        "petMar2_pep      prot     petMar2.pep.fa.gz     NaN      NaN   \n",
        "Myxinidae_pep    prot     Myx.pep.all.fa.gz     NaN  uniprot   \n",
        "braFlo_pep       prot  braFlo.pep.all.fa.gz     NaN  uniprot   \n",
        "\n",
        "                                        terms  \\\n",
        "assembly                                  NaN   \n",
        "danRer_pep                                NaN   \n",
        "homSap_pep                                NaN   \n",
        "musMus_pep                                NaN   \n",
        "petMar2                                   NaN   \n",
        "petMar2_cdna                              NaN   \n",
        "petMar2_cds                               NaN   \n",
        "petMar2_ncrna                             NaN   \n",
        "petMar2_pep                               NaN   \n",
        "Myxinidae_pep                   taxonomy:7762   \n",
        "braFlo_pep     organism:7739+AND+keyword:1185   \n",
        "\n",
        "                                                             url    size  \\\n",
        "assembly       http://athyra.ged.msu.edu/~cswelcher/lamprey/l...  756840   \n",
        "danRer_pep     ftp://ftp.ensembl.org/pub/release-75/fasta/dan...   43153   \n",
        "homSap_pep     ftp://ftp.ensembl.org/pub/release-74/fasta/hom...  104763   \n",
        "musMus_pep     ftp://ftp.ensembl.org/pub/release-75/fasta/mus...   52165   \n",
        "petMar2        ftp://ftp.ensembl.org/pub/release-75/fasta/pet...   25006   \n",
        "petMar2_cdna   ftp://ftp.ensembl.org/pub/release-75/fasta/pet...   11489   \n",
        "petMar2_cds    ftp://ftp.ensembl.org/pub/release-75/fasta/pet...   11442   \n",
        "petMar2_ncrna  ftp://ftp.ensembl.org/pub/release-75/fasta/pet...    2652   \n",
        "petMar2_pep    ftp://ftp.ensembl.org/pub/release-75/fasta/pet...   11442   \n",
        "Myxinidae_pep                                                NaN     990   \n",
        "braFlo_pep                                                   NaN   28544   \n",
        "\n",
        "                              fn  \n",
        "assembly            lamp03.fasta  \n",
        "danRer_pep         danRer.pep.fa  \n",
        "homSap_pep         homSap.pep.fa  \n",
        "musMus_pep         musMus.pep.fa  \n",
        "petMar2               petMar2.fa  \n",
        "petMar2_cdna     petMar2.cdna.fa  \n",
        "petMar2_cds       petMar2.cds.fa  \n",
        "petMar2_ncrna   petMar2.ncrna.fa  \n",
        "petMar2_pep       petMar2.pep.fa  \n",
        "Myxinidae_pep     Myx.pep.all.fa  \n",
        "braFlo_pep     braFlo.pep.all.fa  "
       ]
      }
     ],
     "prompt_number": 28
    },
    {
     "cell_type": "heading",
     "level": 3,
     "metadata": {},
     "source": [
      "Samples"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "sample_metadata = pd.DataFrame(metadata['samples'].dict()).transpose()\n",
      "sample_metadata"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>condition</th>\n",
        "      <th>filename</th>\n",
        "      <th>flowcell</th>\n",
        "      <th>label</th>\n",
        "      <th>paired</th>\n",
        "      <th>tissue</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>614GMAAXX_1_1_pf</th>\n",
        "      <td>          stg18</td>\n",
        "      <td>    614GMAAXX_1_1_pf.qc.fq.gz</td>\n",
        "      <td> 614GM</td>\n",
        "      <td>        embryo_stg18_614GM</td>\n",
        "      <td> False</td>\n",
        "      <td>   embryo</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>614GMAAXX_2_1_pf</th>\n",
        "      <td>          stg20</td>\n",
        "      <td>    614GMAAXX_2_1_pf.qc.fq.gz</td>\n",
        "      <td> 614GM</td>\n",
        "      <td>        embryo_stg20_614GM</td>\n",
        "      <td> False</td>\n",
        "      <td>   embryo</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>614GMAAXX_3_1_pf</th>\n",
        "      <td>         stg22a</td>\n",
        "      <td>    614GMAAXX_3_1_pf.qc.fq.gz</td>\n",
        "      <td> 614GM</td>\n",
        "      <td>       embryo_stg22a_614GM</td>\n",
        "      <td> False</td>\n",
        "      <td>   embryo</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>614GMAAXX_4_1_pf</th>\n",
        "      <td>         stg22b</td>\n",
        "      <td>    614GMAAXX_4_1_pf.qc.fq.gz</td>\n",
        "      <td> 614GM</td>\n",
        "      <td>       embryo_stg22b_614GM</td>\n",
        "      <td> False</td>\n",
        "      <td>   embryo</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>614GMAAXX_6_1_pf</th>\n",
        "      <td>          stg23</td>\n",
        "      <td>    614GMAAXX_6_1_pf.qc.fq.gz</td>\n",
        "      <td> 614GM</td>\n",
        "      <td>        embryo_stg23_614GM</td>\n",
        "      <td> False</td>\n",
        "      <td>   embryo</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>614GMAAXX_7_1_pf</th>\n",
        "      <td>        stg24c1</td>\n",
        "      <td>    614GMAAXX_7_1_pf.qc.fq.gz</td>\n",
        "      <td> 614GM</td>\n",
        "      <td>      embryo_stg24c1_614GM</td>\n",
        "      <td> False</td>\n",
        "      <td>   embryo</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>614GMAAXX_8_1_pf</th>\n",
        "      <td>        stg24c2</td>\n",
        "      <td>    614GMAAXX_8_1_pf.qc.fq.gz</td>\n",
        "      <td> 614GM</td>\n",
        "      <td>      embryo_stg24c2_614GM</td>\n",
        "      <td> False</td>\n",
        "      <td>   embryo</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>AB057JABXX_s_7_pe</th>\n",
        "      <td>               </td>\n",
        "      <td> AB057JABXX_s_7_pe.trim.fq.gz</td>\n",
        "      <td> AB057</td>\n",
        "      <td>               brain_AB057</td>\n",
        "      <td>  True</td>\n",
        "      <td>    brain</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>AI</th>\n",
        "      <td>          adult</td>\n",
        "      <td>                     AI.fq.gz</td>\n",
        "      <td> 614GM</td>\n",
        "      <td>        intest_adult_614GM</td>\n",
        "      <td> False</td>\n",
        "      <td>   intest</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>AK</th>\n",
        "      <td>          adult</td>\n",
        "      <td>                     AK.fq.gz</td>\n",
        "      <td> 614GM</td>\n",
        "      <td>        kidney_adult_614GM</td>\n",
        "      <td> False</td>\n",
        "      <td>   kidney</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>BB0582ABXX_s_3_pe</th>\n",
        "      <td>         larval</td>\n",
        "      <td> BB0582ABXX_s_3_pe.trim.fq.gz</td>\n",
        "      <td> BB058</td>\n",
        "      <td>        liver_larval_BB058</td>\n",
        "      <td>  True</td>\n",
        "      <td>    liver</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>BB0582ABXX_s_4_pe</th>\n",
        "      <td>           pars</td>\n",
        "      <td> BB0582ABXX_s_4_pe.trim.fq.gz</td>\n",
        "      <td> BB058</td>\n",
        "      <td>          liver_pars_BB058</td>\n",
        "      <td>  True</td>\n",
        "      <td>    liver</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>BB0582ABXX_s_5_pe</th>\n",
        "      <td>          adult</td>\n",
        "      <td> BB0582ABXX_s_5_pe.trim.fq.gz</td>\n",
        "      <td> BB058</td>\n",
        "      <td>         brain_adult_BB058</td>\n",
        "      <td>  True</td>\n",
        "      <td>    brain</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>BB0582ABXX_s_6_pe</th>\n",
        "      <td>        adult_1</td>\n",
        "      <td> BB0582ABXX_s_6_pe.trim.fq.gz</td>\n",
        "      <td> BB058</td>\n",
        "      <td>     unknown_adult_1_BB058</td>\n",
        "      <td>  True</td>\n",
        "      <td>  unknown</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>BB0582ABXX_s_7_pe</th>\n",
        "      <td>        adult_2</td>\n",
        "      <td> BB0582ABXX_s_7_pe.trim.fq.gz</td>\n",
        "      <td> BB058</td>\n",
        "      <td>     unknown_adult_2_BB058</td>\n",
        "      <td>  True</td>\n",
        "      <td>  unknown</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>BB0582ABXX_s_8_pe</th>\n",
        "      <td>        adult_3</td>\n",
        "      <td> BB0582ABXX_s_8_pe.trim.fq.gz</td>\n",
        "      <td> BB058</td>\n",
        "      <td>     unknown_adult_3_BB058</td>\n",
        "      <td>  True</td>\n",
        "      <td>  unknown</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FW1-Gill</th>\n",
        "      <td>         freshw</td>\n",
        "      <td>               FW1-Gill.fq.gz</td>\n",
        "      <td> 614H7</td>\n",
        "      <td>         gill_freshw_614H7</td>\n",
        "      <td> False</td>\n",
        "      <td>     gill</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FW1-Intestine</th>\n",
        "      <td>         freshw</td>\n",
        "      <td>          FW1-Intestine.fq.gz</td>\n",
        "      <td> 614H7</td>\n",
        "      <td>       intest_freshw_614H7</td>\n",
        "      <td> False</td>\n",
        "      <td>   intest</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FW1-Kidney</th>\n",
        "      <td>         freshw</td>\n",
        "      <td>             FW1-Kidney.fq.gz</td>\n",
        "      <td> 614H7</td>\n",
        "      <td>       kidney_freshw_614H7</td>\n",
        "      <td> False</td>\n",
        "      <td>   kidney</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>G81-a</th>\n",
        "      <td>  larval_a_tfm0</td>\n",
        "      <td>                  G81-a.fq.gz</td>\n",
        "      <td> 614GT</td>\n",
        "      <td>  gill_larval_a_tfm0_614GT</td>\n",
        "      <td> False</td>\n",
        "      <td>     gill</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>G81-b</th>\n",
        "      <td>  larval_b_tfm0</td>\n",
        "      <td>                  G81-b.fq.gz</td>\n",
        "      <td> 61HWC</td>\n",
        "      <td>  gill_larval_b_tfm0_61HWC</td>\n",
        "      <td> False</td>\n",
        "      <td>     gill</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>G82-a</th>\n",
        "      <td>  larval_a_tfm6</td>\n",
        "      <td>                  G82-a.fq.gz</td>\n",
        "      <td> 614GT</td>\n",
        "      <td>  gill_larval_a_tfm6_614GT</td>\n",
        "      <td> False</td>\n",
        "      <td>     gill</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>G82-b</th>\n",
        "      <td>  larval_b_tfm6</td>\n",
        "      <td>                  G82-b.fq.gz</td>\n",
        "      <td> 61HWC</td>\n",
        "      <td>  gill_larval_b_tfm6_61HWC</td>\n",
        "      <td> False</td>\n",
        "      <td>     gill</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>G83-a</th>\n",
        "      <td> larval_a_tfm12</td>\n",
        "      <td>                  G83-a.fq.gz</td>\n",
        "      <td> 614GT</td>\n",
        "      <td> gill_larval_a_tfm12_614GT</td>\n",
        "      <td> False</td>\n",
        "      <td>     gill</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>G83-b</th>\n",
        "      <td> larval_b_tfm12</td>\n",
        "      <td>                  G83-b.fq.gz</td>\n",
        "      <td> 61HWC</td>\n",
        "      <td> gill_larval_b_tfm12_61HWC</td>\n",
        "      <td> False</td>\n",
        "      <td>     gill</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Gill</th>\n",
        "      <td>               </td>\n",
        "      <td>                   Gill.fq.gz</td>\n",
        "      <td> 42UL4</td>\n",
        "      <td>                gill_42UL4</td>\n",
        "      <td> False</td>\n",
        "      <td>     gill</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>JV1_liver</th>\n",
        "      <td>            juv</td>\n",
        "      <td>              JV1_liver.fq.gz</td>\n",
        "      <td> 42UKJ</td>\n",
        "      <td>          intest_juv_42UKJ</td>\n",
        "      <td> False</td>\n",
        "      <td>   intest</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>JV_intest</th>\n",
        "      <td>          juv_b</td>\n",
        "      <td>              JV_intest.fq.gz</td>\n",
        "      <td> 42UKJ</td>\n",
        "      <td>         liver_juv_b_42UKJ</td>\n",
        "      <td> False</td>\n",
        "      <td>    liver</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>JVa_liver</th>\n",
        "      <td>          juv_1</td>\n",
        "      <td>              JVa_liver.fq.gz</td>\n",
        "      <td> 42UKJ</td>\n",
        "      <td>         liver_juv_1_42UKJ</td>\n",
        "      <td> False</td>\n",
        "      <td>    liver</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>JVb_liver</th>\n",
        "      <td>          juv_a</td>\n",
        "      <td>              JVb_liver.fq.gz</td>\n",
        "      <td> 42UKJ</td>\n",
        "      <td>         liver_juv_a_42UKJ</td>\n",
        "      <td> False</td>\n",
        "      <td>    liver</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>...</th>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SMHS</th>\n",
        "      <td>        m_sperm</td>\n",
        "      <td>                   SMHS.fq.gz</td>\n",
        "      <td> 61504</td>\n",
        "      <td>    headskin_m_sperm_61504</td>\n",
        "      <td> False</td>\n",
        "      <td> headskin</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SPDI</th>\n",
        "      <td>    distal_pars</td>\n",
        "      <td>                   SPDI.fq.gz</td>\n",
        "      <td> 434JM</td>\n",
        "      <td>  intest_distal_pars_434JM</td>\n",
        "      <td> False</td>\n",
        "      <td>   intest</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SPK</th>\n",
        "      <td>           pars</td>\n",
        "      <td>                    SPK.fq.gz</td>\n",
        "      <td> 434JM</td>\n",
        "      <td>         kidney_pars_434JM</td>\n",
        "      <td> False</td>\n",
        "      <td>   kidney</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SPPI</th>\n",
        "      <td>      prox_pars</td>\n",
        "      <td>                   SPPI.fq.gz</td>\n",
        "      <td> 434JM</td>\n",
        "      <td>    intest_prox_pars_434JM</td>\n",
        "      <td> False</td>\n",
        "      <td>   intest</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SW2-Gill</th>\n",
        "      <td>          saltw</td>\n",
        "      <td>               SW2-Gill.fq.gz</td>\n",
        "      <td> 614H7</td>\n",
        "      <td>          gill_saltw_614H7</td>\n",
        "      <td> False</td>\n",
        "      <td>     gill</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SW2-Intestine-a</th>\n",
        "      <td>        saltw_a</td>\n",
        "      <td>        SW2-Intestine-a.fq.gz</td>\n",
        "      <td> 614H7</td>\n",
        "      <td>      intest_saltw_a_614H7</td>\n",
        "      <td> False</td>\n",
        "      <td>   intest</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SW2-Intestine-b</th>\n",
        "      <td>        saltw_b</td>\n",
        "      <td>        SW2-Intestine-b.fq.gz</td>\n",
        "      <td> 614H7</td>\n",
        "      <td>      intest_saltw_b_614H7</td>\n",
        "      <td> False</td>\n",
        "      <td>   intest</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SW2-Kidney</th>\n",
        "      <td>          saltw</td>\n",
        "      <td>             SW2-Kidney.fq.gz</td>\n",
        "      <td> 614H7</td>\n",
        "      <td>        kidney_saltw_614H7</td>\n",
        "      <td> False</td>\n",
        "      <td>   kidney</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Stg-1-Intestine</th>\n",
        "      <td>           stg1</td>\n",
        "      <td>        Stg-1-Intestine.fq.gz</td>\n",
        "      <td> 614VD</td>\n",
        "      <td>         intest_stg1_614VD</td>\n",
        "      <td> False</td>\n",
        "      <td>   intest</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Stg-1-Kidney</th>\n",
        "      <td>           stg1</td>\n",
        "      <td>           Stg-1-Kidney.fq.gz</td>\n",
        "      <td> 614VE</td>\n",
        "      <td>         kidney_stg1_614VE</td>\n",
        "      <td> False</td>\n",
        "      <td>   kidney</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Stg-2-Intestine</th>\n",
        "      <td>           stg2</td>\n",
        "      <td>        Stg-2-Intestine.fq.gz</td>\n",
        "      <td> 614VD</td>\n",
        "      <td>         intest_stg2_614VD</td>\n",
        "      <td> False</td>\n",
        "      <td>   intest</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Stg-2-Kidney</th>\n",
        "      <td>           stg2</td>\n",
        "      <td>           Stg-2-Kidney.fq.gz</td>\n",
        "      <td> 614VE</td>\n",
        "      <td>         kidney_stg2_614VE</td>\n",
        "      <td> False</td>\n",
        "      <td>   kidney</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Stg-3-Intestine</th>\n",
        "      <td>           stg3</td>\n",
        "      <td>        Stg-3-Intestine.fq.gz</td>\n",
        "      <td> 614VD</td>\n",
        "      <td>         intest_stg3_614VD</td>\n",
        "      <td> False</td>\n",
        "      <td>   intest</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Stg-3-Kidney</th>\n",
        "      <td>           stg3</td>\n",
        "      <td>           Stg-3-Kidney.fq.gz</td>\n",
        "      <td> 614VE</td>\n",
        "      <td>         kidney_stg3_614VE</td>\n",
        "      <td> False</td>\n",
        "      <td>   kidney</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Stg-4-Intestine</th>\n",
        "      <td>           stg4</td>\n",
        "      <td>        Stg-4-Intestine.fq.gz</td>\n",
        "      <td> 614VD</td>\n",
        "      <td>         intest_stg4_614VD</td>\n",
        "      <td> False</td>\n",
        "      <td>   intest</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Stg-4-Kidney</th>\n",
        "      <td>           stg4</td>\n",
        "      <td>           Stg-4-Kidney.fq.gz</td>\n",
        "      <td> 614VE</td>\n",
        "      <td>         kidney_stg4_614VE</td>\n",
        "      <td> False</td>\n",
        "      <td>   kidney</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Stg-5-Intestine</th>\n",
        "      <td>           stg5</td>\n",
        "      <td>        Stg-5-Intestine.fq.gz</td>\n",
        "      <td> 614VD</td>\n",
        "      <td>         intest_stg5_614VD</td>\n",
        "      <td> False</td>\n",
        "      <td>   intest</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Stg-5-Kidney</th>\n",
        "      <td>           stg5</td>\n",
        "      <td>           Stg-5-Kidney.fq.gz</td>\n",
        "      <td> 614VE</td>\n",
        "      <td>         kidney_stg5_614VE</td>\n",
        "      <td> False</td>\n",
        "      <td>   kidney</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Stg-6-Intestine</th>\n",
        "      <td>           stg6</td>\n",
        "      <td>        Stg-6-Intestine.fq.gz</td>\n",
        "      <td> 614VD</td>\n",
        "      <td>         intest_stg6_614VD</td>\n",
        "      <td> False</td>\n",
        "      <td>   intest</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Stg-6-Kidney</th>\n",
        "      <td>           stg6</td>\n",
        "      <td>           Stg-6-Kidney.fq.gz</td>\n",
        "      <td> 614VE</td>\n",
        "      <td>         kidney_stg6_614VE</td>\n",
        "      <td> False</td>\n",
        "      <td>   kidney</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Stg-7-Intestine</th>\n",
        "      <td>           stg7</td>\n",
        "      <td>        Stg-7-Intestine.fq.gz</td>\n",
        "      <td> 614VD</td>\n",
        "      <td>         intest_stg7_614VD</td>\n",
        "      <td> False</td>\n",
        "      <td>   intest</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Stg-7-Kidney</th>\n",
        "      <td>           stg7</td>\n",
        "      <td>           Stg-7-Kidney.fq.gz</td>\n",
        "      <td> 614VE</td>\n",
        "      <td>         kidney_stg7_614VE</td>\n",
        "      <td> False</td>\n",
        "      <td>   kidney</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>brain_paired</th>\n",
        "      <td>               </td>\n",
        "      <td>        brain_paired.qc.fq.gz</td>\n",
        "      <td> 30M2H</td>\n",
        "      <td>               brain_30M2H</td>\n",
        "      <td>  True</td>\n",
        "      <td>    brain</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>mssm_single_s1</th>\n",
        "      <td>           ctrl</td>\n",
        "      <td>    mssm_single_s1.trim.fq.gz</td>\n",
        "      <td>  mssm</td>\n",
        "      <td>           brain_ctrl_mssm</td>\n",
        "      <td> False</td>\n",
        "      <td>    brain</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>mssm_single_s2</th>\n",
        "      <td>           ctrl</td>\n",
        "      <td>    mssm_single_s2.trim.fq.gz</td>\n",
        "      <td>  mssm</td>\n",
        "      <td>          spinal_ctrl_mssm</td>\n",
        "      <td> False</td>\n",
        "      <td>   spinal</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>mssm_single_s3</th>\n",
        "      <td>           day3</td>\n",
        "      <td>    mssm_single_s3.trim.fq.gz</td>\n",
        "      <td>  mssm</td>\n",
        "      <td>           brain_day3_mssm</td>\n",
        "      <td> False</td>\n",
        "      <td>    brain</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>mssm_single_s5</th>\n",
        "      <td>           day3</td>\n",
        "      <td>    mssm_single_s5.trim.fq.gz</td>\n",
        "      <td>  mssm</td>\n",
        "      <td>          spinal_day3_mssm</td>\n",
        "      <td> False</td>\n",
        "      <td>   spinal</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>mssm_single_s6</th>\n",
        "      <td>          day21</td>\n",
        "      <td>    mssm_single_s6.trim.fq.gz</td>\n",
        "      <td>  mssm</td>\n",
        "      <td>          brain_day21_mssm</td>\n",
        "      <td> False</td>\n",
        "      <td>    brain</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>mssm_single_s7</th>\n",
        "      <td>          day21</td>\n",
        "      <td>    mssm_single_s7.trim.fq.gz</td>\n",
        "      <td>  mssm</td>\n",
        "      <td>         spinal_day21_mssm</td>\n",
        "      <td> False</td>\n",
        "      <td>   spinal</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>spinal_cord_paired</th>\n",
        "      <td>               </td>\n",
        "      <td>                             </td>\n",
        "      <td> 3102H</td>\n",
        "      <td>              spinal_3102H</td>\n",
        "      <td>  True</td>\n",
        "      <td>   spinal</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>84 rows \u00d7 6 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 64,
       "text": [
        "                         condition                      filename flowcell  \\\n",
        "614GMAAXX_1_1_pf             stg18     614GMAAXX_1_1_pf.qc.fq.gz    614GM   \n",
        "614GMAAXX_2_1_pf             stg20     614GMAAXX_2_1_pf.qc.fq.gz    614GM   \n",
        "614GMAAXX_3_1_pf            stg22a     614GMAAXX_3_1_pf.qc.fq.gz    614GM   \n",
        "614GMAAXX_4_1_pf            stg22b     614GMAAXX_4_1_pf.qc.fq.gz    614GM   \n",
        "614GMAAXX_6_1_pf             stg23     614GMAAXX_6_1_pf.qc.fq.gz    614GM   \n",
        "614GMAAXX_7_1_pf           stg24c1     614GMAAXX_7_1_pf.qc.fq.gz    614GM   \n",
        "614GMAAXX_8_1_pf           stg24c2     614GMAAXX_8_1_pf.qc.fq.gz    614GM   \n",
        "AB057JABXX_s_7_pe                   AB057JABXX_s_7_pe.trim.fq.gz    AB057   \n",
        "AI                           adult                      AI.fq.gz    614GM   \n",
        "AK                           adult                      AK.fq.gz    614GM   \n",
        "BB0582ABXX_s_3_pe           larval  BB0582ABXX_s_3_pe.trim.fq.gz    BB058   \n",
        "BB0582ABXX_s_4_pe             pars  BB0582ABXX_s_4_pe.trim.fq.gz    BB058   \n",
        "BB0582ABXX_s_5_pe            adult  BB0582ABXX_s_5_pe.trim.fq.gz    BB058   \n",
        "BB0582ABXX_s_6_pe          adult_1  BB0582ABXX_s_6_pe.trim.fq.gz    BB058   \n",
        "BB0582ABXX_s_7_pe          adult_2  BB0582ABXX_s_7_pe.trim.fq.gz    BB058   \n",
        "BB0582ABXX_s_8_pe          adult_3  BB0582ABXX_s_8_pe.trim.fq.gz    BB058   \n",
        "FW1-Gill                    freshw                FW1-Gill.fq.gz    614H7   \n",
        "FW1-Intestine               freshw           FW1-Intestine.fq.gz    614H7   \n",
        "FW1-Kidney                  freshw              FW1-Kidney.fq.gz    614H7   \n",
        "G81-a                larval_a_tfm0                   G81-a.fq.gz    614GT   \n",
        "G81-b                larval_b_tfm0                   G81-b.fq.gz    61HWC   \n",
        "G82-a                larval_a_tfm6                   G82-a.fq.gz    614GT   \n",
        "G82-b                larval_b_tfm6                   G82-b.fq.gz    61HWC   \n",
        "G83-a               larval_a_tfm12                   G83-a.fq.gz    614GT   \n",
        "G83-b               larval_b_tfm12                   G83-b.fq.gz    61HWC   \n",
        "Gill                                                  Gill.fq.gz    42UL4   \n",
        "JV1_liver                      juv               JV1_liver.fq.gz    42UKJ   \n",
        "JV_intest                    juv_b               JV_intest.fq.gz    42UKJ   \n",
        "JVa_liver                    juv_1               JVa_liver.fq.gz    42UKJ   \n",
        "JVb_liver                    juv_a               JVb_liver.fq.gz    42UKJ   \n",
        "...                            ...                           ...      ...   \n",
        "SMHS                       m_sperm                    SMHS.fq.gz    61504   \n",
        "SPDI                   distal_pars                    SPDI.fq.gz    434JM   \n",
        "SPK                           pars                     SPK.fq.gz    434JM   \n",
        "SPPI                     prox_pars                    SPPI.fq.gz    434JM   \n",
        "SW2-Gill                     saltw                SW2-Gill.fq.gz    614H7   \n",
        "SW2-Intestine-a            saltw_a         SW2-Intestine-a.fq.gz    614H7   \n",
        "SW2-Intestine-b            saltw_b         SW2-Intestine-b.fq.gz    614H7   \n",
        "SW2-Kidney                   saltw              SW2-Kidney.fq.gz    614H7   \n",
        "Stg-1-Intestine               stg1         Stg-1-Intestine.fq.gz    614VD   \n",
        "Stg-1-Kidney                  stg1            Stg-1-Kidney.fq.gz    614VE   \n",
        "Stg-2-Intestine               stg2         Stg-2-Intestine.fq.gz    614VD   \n",
        "Stg-2-Kidney                  stg2            Stg-2-Kidney.fq.gz    614VE   \n",
        "Stg-3-Intestine               stg3         Stg-3-Intestine.fq.gz    614VD   \n",
        "Stg-3-Kidney                  stg3            Stg-3-Kidney.fq.gz    614VE   \n",
        "Stg-4-Intestine               stg4         Stg-4-Intestine.fq.gz    614VD   \n",
        "Stg-4-Kidney                  stg4            Stg-4-Kidney.fq.gz    614VE   \n",
        "Stg-5-Intestine               stg5         Stg-5-Intestine.fq.gz    614VD   \n",
        "Stg-5-Kidney                  stg5            Stg-5-Kidney.fq.gz    614VE   \n",
        "Stg-6-Intestine               stg6         Stg-6-Intestine.fq.gz    614VD   \n",
        "Stg-6-Kidney                  stg6            Stg-6-Kidney.fq.gz    614VE   \n",
        "Stg-7-Intestine               stg7         Stg-7-Intestine.fq.gz    614VD   \n",
        "Stg-7-Kidney                  stg7            Stg-7-Kidney.fq.gz    614VE   \n",
        "brain_paired                               brain_paired.qc.fq.gz    30M2H   \n",
        "mssm_single_s1                ctrl     mssm_single_s1.trim.fq.gz     mssm   \n",
        "mssm_single_s2                ctrl     mssm_single_s2.trim.fq.gz     mssm   \n",
        "mssm_single_s3                day3     mssm_single_s3.trim.fq.gz     mssm   \n",
        "mssm_single_s5                day3     mssm_single_s5.trim.fq.gz     mssm   \n",
        "mssm_single_s6               day21     mssm_single_s6.trim.fq.gz     mssm   \n",
        "mssm_single_s7               day21     mssm_single_s7.trim.fq.gz     mssm   \n",
        "spinal_cord_paired                                                  3102H   \n",
        "\n",
        "                                        label paired    tissue  \n",
        "614GMAAXX_1_1_pf           embryo_stg18_614GM  False    embryo  \n",
        "614GMAAXX_2_1_pf           embryo_stg20_614GM  False    embryo  \n",
        "614GMAAXX_3_1_pf          embryo_stg22a_614GM  False    embryo  \n",
        "614GMAAXX_4_1_pf          embryo_stg22b_614GM  False    embryo  \n",
        "614GMAAXX_6_1_pf           embryo_stg23_614GM  False    embryo  \n",
        "614GMAAXX_7_1_pf         embryo_stg24c1_614GM  False    embryo  \n",
        "614GMAAXX_8_1_pf         embryo_stg24c2_614GM  False    embryo  \n",
        "AB057JABXX_s_7_pe                 brain_AB057   True     brain  \n",
        "AI                         intest_adult_614GM  False    intest  \n",
        "AK                         kidney_adult_614GM  False    kidney  \n",
        "BB0582ABXX_s_3_pe          liver_larval_BB058   True     liver  \n",
        "BB0582ABXX_s_4_pe            liver_pars_BB058   True     liver  \n",
        "BB0582ABXX_s_5_pe           brain_adult_BB058   True     brain  \n",
        "BB0582ABXX_s_6_pe       unknown_adult_1_BB058   True   unknown  \n",
        "BB0582ABXX_s_7_pe       unknown_adult_2_BB058   True   unknown  \n",
        "BB0582ABXX_s_8_pe       unknown_adult_3_BB058   True   unknown  \n",
        "FW1-Gill                    gill_freshw_614H7  False      gill  \n",
        "FW1-Intestine             intest_freshw_614H7  False    intest  \n",
        "FW1-Kidney                kidney_freshw_614H7  False    kidney  \n",
        "G81-a                gill_larval_a_tfm0_614GT  False      gill  \n",
        "G81-b                gill_larval_b_tfm0_61HWC  False      gill  \n",
        "G82-a                gill_larval_a_tfm6_614GT  False      gill  \n",
        "G82-b                gill_larval_b_tfm6_61HWC  False      gill  \n",
        "G83-a               gill_larval_a_tfm12_614GT  False      gill  \n",
        "G83-b               gill_larval_b_tfm12_61HWC  False      gill  \n",
        "Gill                               gill_42UL4  False      gill  \n",
        "JV1_liver                    intest_juv_42UKJ  False    intest  \n",
        "JV_intest                   liver_juv_b_42UKJ  False     liver  \n",
        "JVa_liver                   liver_juv_1_42UKJ  False     liver  \n",
        "JVb_liver                   liver_juv_a_42UKJ  False     liver  \n",
        "...                                       ...    ...       ...  \n",
        "SMHS                   headskin_m_sperm_61504  False  headskin  \n",
        "SPDI                 intest_distal_pars_434JM  False    intest  \n",
        "SPK                         kidney_pars_434JM  False    kidney  \n",
        "SPPI                   intest_prox_pars_434JM  False    intest  \n",
        "SW2-Gill                     gill_saltw_614H7  False      gill  \n",
        "SW2-Intestine-a          intest_saltw_a_614H7  False    intest  \n",
        "SW2-Intestine-b          intest_saltw_b_614H7  False    intest  \n",
        "SW2-Kidney                 kidney_saltw_614H7  False    kidney  \n",
        "Stg-1-Intestine             intest_stg1_614VD  False    intest  \n",
        "Stg-1-Kidney                kidney_stg1_614VE  False    kidney  \n",
        "Stg-2-Intestine             intest_stg2_614VD  False    intest  \n",
        "Stg-2-Kidney                kidney_stg2_614VE  False    kidney  \n",
        "Stg-3-Intestine             intest_stg3_614VD  False    intest  \n",
        "Stg-3-Kidney                kidney_stg3_614VE  False    kidney  \n",
        "Stg-4-Intestine             intest_stg4_614VD  False    intest  \n",
        "Stg-4-Kidney                kidney_stg4_614VE  False    kidney  \n",
        "Stg-5-Intestine             intest_stg5_614VD  False    intest  \n",
        "Stg-5-Kidney                kidney_stg5_614VE  False    kidney  \n",
        "Stg-6-Intestine             intest_stg6_614VD  False    intest  \n",
        "Stg-6-Kidney                kidney_stg6_614VE  False    kidney  \n",
        "Stg-7-Intestine             intest_stg7_614VD  False    intest  \n",
        "Stg-7-Kidney                kidney_stg7_614VE  False    kidney  \n",
        "brain_paired                      brain_30M2H   True     brain  \n",
        "mssm_single_s1                brain_ctrl_mssm  False     brain  \n",
        "mssm_single_s2               spinal_ctrl_mssm  False    spinal  \n",
        "mssm_single_s3                brain_day3_mssm  False     brain  \n",
        "mssm_single_s5               spinal_day3_mssm  False    spinal  \n",
        "mssm_single_s6               brain_day21_mssm  False     brain  \n",
        "mssm_single_s7              spinal_day21_mssm  False    spinal  \n",
        "spinal_cord_paired               spinal_3102H   True    spinal  \n",
        "\n",
        "[84 rows x 6 columns]"
       ]
      }
     ],
     "prompt_number": 64
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Data"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Create an HDF5 volume to store data needed in the other notebooks; we won't be performing ops directly on it, so use maximum compression to save disk space. "
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "store = pd.HDFStore(wdir('{}.store.h5'.format(metadata['prefix'])), complib='zlib', complevel=9)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 75
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "store['db_metadata'] = db_metadata"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 76
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "store['sample_metadata'] = sample_metadata"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 77
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "screed.read_fasta_sequences(wdir(db_metadata.fn['assembly']))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 33,
       "text": [
        "<ScreedDB, '../../work/lamp03.fasta_screed'>"
       ]
      }
     ],
     "prompt_number": 33
    },
    {
     "cell_type": "heading",
     "level": 3,
     "metadata": {},
     "source": [
      "Transcript Support"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "genes_rsem_df = pd.read_csv(wdir('all.lamp3.rsem.matrix'), delimiter='\\t', index_col=0)\n",
      "iso_rsem_df = pd.read_csv(wdir('all.lamp3.iso.rsem.matrix'), delimiter='\\t', index_col=0)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 34
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "genes_support_df = pd.DataFrame(index=genes_rsem_df.index)\n",
      "iso_support_df = pd.DataFrame(index=iso_rsem_df.index)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 35
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for col in iso_rsem_df:\n",
      "    iso_support_df[col] = 0\n",
      "    iso_support_df.loc[iso_rsem_df[col] > 1.0, col] = 1"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 36
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for col in genes_rsem_df:\n",
      "    genes_support_df[col] = 0\n",
      "    genes_support_df.loc[genes_rsem_df[col] > 1.0, col] = 1"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 37
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "labels = {k:v['label'] for k, v in metadata['samples'].items()}"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 38
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "iso_support_df.rename(columns=labels, inplace=True)\n",
      "iso_rsem_df.rename(columns=labels, inplace=True)\n",
      "genes_support_df.rename(columns=labels, inplace=True)\n",
      "genes_rsem_df.rename(columns=labels, inplace=True)\n",
      "\n",
      "iso_support_df.sort_index(axis=1, inplace=True)\n",
      "iso_rsem_df.sort_index(axis=1, inplace=True)\n",
      "genes_support_df.sort_index(axis=1, inplace=True)\n",
      "genes_rsem_df.sort_index(axis=1, inplace=True)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 39
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "labels_df = pd.DataFrame(labels, index=['newname']).transpose()\n",
      "labels_df['tissue'] = labels_df.newname.apply(lambda x: x.partition('_')[0])\n",
      "labels_df['tissue_c'] = Categorical(list(labels_df.tissue)).labels\n",
      "labels_df.sort(columns='newname', inplace=True)\n",
      "tissue_palette = sns.color_palette('Set3', n_colors=len(labels_df.groupby('tissue_c').count()))\n",
      "labels_df['color'] = labels_df.tissue_c.apply(lambda x: tissue_palette[x])"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 44
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "We know this sample is contaminated; so, we'll mask any transcripts unique to that sample."
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "contaminants = (iso_support_df.sum(axis=1) == 1) & iso_support_df.spinal_3102H"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 52
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "iso_support_df = iso_support_df[contaminants == False]\n",
      "iso_rsem_df = iso_rsem_df[contaminants == False]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 53
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "store['iso_support_df'] = iso_support_df\n",
      "store['iso_rsem_df'] = iso_rsem_df\n",
      "store['labels_df'] = labels_df"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 78
    },
    {
     "cell_type": "heading",
     "level": 3,
     "metadata": {},
     "source": [
      "Blast Results"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "blast_items = [(n, info) for n, info in metadata['urls'].items() + metadata['queries'].items() if n != 'assembly']\n",
      "for i, (dbname, info) in enumerate(blast_items):\n",
      "    target = '{}.fasta.x.{}.db.tsv'.format(metadata['prefix'], info['dest'].rstrip('.gz'))\n",
      "    print dbname, target\n",
      "    tmp = pd.merge(pd.DataFrame(index=iso_support_df.index),\n",
      "                      pd.read_table(wdir(target),\n",
      "                      header=None, index_col=0, names=outfmt6),\n",
      "                      left_index=True, right_index=True, how='left')\n",
      "    tmp['ind'] = tmp.index\n",
      "    tmp.sort(columns=['ind', 'evalue'])\n",
      "    tmp.drop_duplicates(cols='ind', inplace=True)\n",
      "    del tmp['ind']\n",
      "    if i == 0:\n",
      "        blast_panel = pd.Panel({dbname: tmp})\n",
      "    else:\n",
      "        blast_panel[dbname] = tmp"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "petMar2 lamp03.fasta.x.petMar2.fa.db.tsv\n",
        "petMar2_pep"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        " lamp03.fasta.x.petMar2.pep.fa.db.tsv\n",
        "petMar2_cds"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        " lamp03.fasta.x.petMar2.cds.fa.db.tsv\n",
        "petMar2_ncrna"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        " lamp03.fasta.x.petMar2.ncrna.fa.db.tsv\n",
        "petMar2_cdna"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        " lamp03.fasta.x.petMar2.cdna.fa.db.tsv\n",
        "musMus_pep"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        " lamp03.fasta.x.musMus.pep.fa.db.tsv\n",
        "danRer_pep"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        " lamp03.fasta.x.danRer.pep.fa.db.tsv\n",
        "homSap_pep"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        " lamp03.fasta.x.homSap.pep.fa.db.tsv\n",
        "braFlo_pep"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        " lamp03.fasta.x.braFlo.pep.all.fa.db.tsv\n",
        "Myxinidae_pep"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        " lamp03.fasta.x.Myx.pep.all.fa.db.tsv\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "/home/camille/anaconda/envs/bio/lib/python2.7/site-packages/pandas/util/decorators.py:53: FutureWarning: cols is deprecated, use subset instead\n",
        "  warnings.warn(msg, FutureWarning)\n"
       ]
      }
     ],
     "prompt_number": 55
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "blast_items = [(n, info) for n, info in metadata['urls'].items() + metadata['queries'].items() if n != 'assembly']\n",
      "for i, (dbname, info) in enumerate(blast_items):\n",
      "    A_fn = '{}.fasta.x.{}.db.tsv'.format(metadata['prefix'], info['dest'].rstrip('.gz'))\n",
      "    B_fn = '{}.db.x.{}.fasta.tsv'.format(info['dest'].rstrip('.gz'), metadata['prefix'])\n",
      "    \n",
      "    print '{} <=> {}'.format(A_fn, B_fn)\n",
      "    \n",
      "    A = pd.read_table(wdir(A_fn), header=None, index_col=0, names=outfmt6)\n",
      "    B = pd.read_table(wdir(B_fn), header=None, index_col=0, names=outfmt6)\n",
      "\n",
      "    A['ind'] = A.index\n",
      "    A.sort(columns=['ind', 'evalue'])\n",
      "    A.drop_duplicates(cols='ind', inplace=True)\n",
      "    del A['ind']\n",
      "    \n",
      "    B['ind'] = B.index\n",
      "    B.sort(columns=['ind', 'evalue'])\n",
      "    B.drop_duplicates(cols='ind', inplace=True)\n",
      "    del B['ind']\n",
      "    \n",
      "    X = pd.merge(A, B, how='inner', left_on='sseqid', right_index=True)\n",
      "    \n",
      "    del X['sseqid_x']\n",
      "    del X['sseqid']\n",
      "    \n",
      "    X = pd.merge(pd.DataFrame(index=iso_support_df.index),\n",
      "                      X[(X.index == X.sseqid_y)], left_index=True, right_index=True, how='left')\n",
      "    X.rename({'sseqid_y': 'sseqid'}, inplace=True)\n",
      "    \n",
      "    if i == 0:\n",
      "        ortho_panel = pd.Panel({dbname: X})\n",
      "    else:\n",
      "        ortho_panel[dbname] = X"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "lamp03.fasta.x.petMar2.fa.db.tsv <=> petMar2.fa.db.x.lamp03.fasta.tsv\n",
        "lamp03.fasta.x.petMar2.pep.fa.db.tsv <=> petMar2.pep.fa.db.x.lamp03.fasta.tsv"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "lamp03.fasta.x.petMar2.cds.fa.db.tsv <=> petMar2.cds.fa.db.x.lamp03.fasta.tsv"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "lamp03.fasta.x.petMar2.ncrna.fa.db.tsv <=> petMar2.ncrna.fa.db.x.lamp03.fasta.tsv"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "lamp03.fasta.x.petMar2.cdna.fa.db.tsv <=> petMar2.cdna.fa.db.x.lamp03.fasta.tsv"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "lamp03.fasta.x.musMus.pep.fa.db.tsv <=> musMus.pep.fa.db.x.lamp03.fasta.tsv"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "lamp03.fasta.x.danRer.pep.fa.db.tsv <=> danRer.pep.fa.db.x.lamp03.fasta.tsv"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "lamp03.fasta.x.homSap.pep.fa.db.tsv <=> homSap.pep.fa.db.x.lamp03.fasta.tsv"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "lamp03.fasta.x.braFlo.pep.all.fa.db.tsv <=> braFlo.pep.all.fa.db.x.lamp03.fasta.tsv"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "lamp03.fasta.x.Myx.pep.all.fa.db.tsv <=> Myx.pep.all.fa.db.x.lamp03.fasta.tsv"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n"
       ]
      }
     ],
     "prompt_number": 56
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "blast_df = blast_panel.minor_xs('evalue') >= 0"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 57
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "ortho_df = ortho_panel.minor_xs('evalue_x') >= 0"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 58
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "putative_df = pd.DataFrame({'homology': (blast_df.sum(axis=1) > 0)})\n",
      "putative_df['orthology'] = (ortho_df.sum(axis=1) > 0)\n",
      "putative_df['not_lamp00'] = ((blast_df.petMar2_cdna | blast_df.petMar2_cds | \\\n",
      "                             blast_df.petMar2_pep | blast_df.petMar2_ncrna) == False)\n",
      "putative_df['novel'] = ((blast_df.petMar2 == False) & putative_df.not_lamp00)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 59
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "tissue_tr_df = iso_support_df.groupby(by=labels_df.tissue.values, axis=1).sum()\n",
      "tissue_support_df = iso_support_df.groupby(by=labels_df.tissue.values, axis=1).sum() > 0"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 74
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "store['tissue_tr_df'] = tissue_tr_df\n",
      "store['tissue_support_df'] = tissue_support_df"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 79
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "store['blast_df'] = blast_df\n",
      "store['ortho_df'] = ortho_df"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 80
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "store['blast_panel'] = blast_panel\n",
      "store['ortho_panel'] = ortho_panel"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 81
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "store['putative_df'] = putative_df"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 82
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "store.close()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 83
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    }
   ],
   "metadata": {}
  }
 ]
}